* Set serial console baudrate to 115200 since cable can't handle 1500000
* Enable UART3
* Enable SPI3
* Disable HDMI or there will be a pin conflict for SPI (for some reason)

Upstream-Status: Inappropriate

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi
index 1ee5d96a46a1..dfda7742e1e6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi
@@ -7,7 +7,7 @@
 
 / {
 	chosen {
-		stdout-path = "serial2:1500000n8";
+		stdout-path = "serial2:115200n8";
 	};
 
 	hdmi-con {
@@ -185,7 +185,7 @@ &gpu {
 &hdmi {
 	avdd-0v9-supply = <&vdda_0v9>;
 	avdd-1v8-supply = <&vcca1v8_image>;
-	status = "okay";
+	status = "disabled";
 };
 
 &hdmi_in {
@@ -492,6 +492,23 @@ &uart2 {
 	status = "okay";
 };
 
+&uart3 {
+	status = "okay";
+};
+
+&spi3 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	pinctrl-0 = <&spi3m1_cs0_hs &spi3m1_cs1_hs &spi3m1_pins_hs>;
+	status = "okay";
+
+	spidev: spidev@0 {
+		compatible = "unitexe,spi-any";
+		spi-max-frequency = <50000000>;
+		reg = <0>;
+	};
+};
+
 &usb_host0_xhci {
 	status = "okay";
 };
